Python/ceval.c

```diff
@@ -4686,7 +4686,9 @@ _PyEval_EvalFrameDefault(PyThreadState *tstate, _PyInterpreterFrame *frame, int
             function = PEEK(total_args + 1);
             int positional_args = total_args - KWNAMES_LEN();
             // Check if the call can be inlined or not
-            if (Py_TYPE(function) == &PyFunction_Type && tstate->interp->eval_frame == NULL) {
+            if (Py_TYPE(function) == &PyFunction_Type &&
+                tstate->interp->eval_frame == NULL &&
+                ((PyFunctionObject *)function)->vectorcall == _PyFunction_Vectorcall) {
```

Comment:
I have a personal preference for this coding style:

```suggestion
            if (Py_TYPE(function) == &PyFunction_Type
                && tstate->interp->eval_frame == NULL
                && ((PyFunctionObject *)function)->vectorcall == _PyFunction_Vectorcall) {
```

Reply:
The author's code follows PEP 7 https://peps.python.org/pep-0007/#code-lay-out.